app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js

```diff
@@ -77,14 +77,36 @@ class SectionLink {
     return this.name === undefined || this.validName ? "" : "warning";
   }
 
+  get internal() {
+    return (
+      this.value.startsWith(this.protocolAndHost) || this.value.startsWith("/")
+    );
+  }
+
+  get validExternal() {
+    if (this.internal) {
+      return false;
+    }
+    try {
+      return new URL(this.value);
+    } catch {
+      return false;
+    }
+  }
+
+  get validInternal() {
```

Comment:
```suggestion
  #validInternal() {
```

Same as https://github.com/discourse/discourse/pull/20503/files#r1122596388